### ISSUE TITLE
ReaderFactory.Open does not check for SevenZipArchive

### ISSUE THREAD

**Issue by matthewDDennis** [adamhathcock/sharpcompress#491]
Thus when using ReaderFactory.Open I get an error of

> Cannot determine compressed stream type. Supported Reader Formats: Zip, GZip, BZip2, Tar, Rar, LZip, XZ

**Comment by adamhathcock** [comment]
7Zip isn’t supported as a Reader.  7Zip the format sucks because you can’t stream it.  You need random file access to properly extract data.   You have to use ArchiveFactory.

**Comment by matthewDDennis** [comment]
Yeah, I figured it out, but had to read the source code. Are there docs?

**Comment by adamhathcock** [comment]
Linked from the readme: https://github.com/adamhathcock/sharpcompress/blob/master/FORMATS.md
